src/main/scala/fs2/kafka/KafkaConsumer.scala

```diff
@@ -358,7 +360,10 @@ private[kafka] object KafkaConsumer {
         actorFiber combine pollsFiber
       }
 
+      private val streamIdCounter = new AtomicInteger(0)
+
       override def partitionedStream: Stream[F, Stream[F, CommittableMessage[F, K, V]]] = {
+        val streamId = streamIdCounter.getAndIncrement()
```

Comment:
Unfortunately, this breaks referential transparency. `Ref[F, Int]` and `modify` is the alternative.

Reply:
How do you plan to use `Ref[F, Int]` to avoid breaking referential transparency while not changing the interface?

Reply:
We create the `Ref[F, Int]` as part of `consumerResource`, and pass it to `createKafkaConsumer`. In `partitionedStream` we then add something like the following.

```scala
Stream.eval(ref.modify(n => (n + 1, n))).flatMap { streamId =>
  // ...
}
```